### PR TITLE
Database hot standby

### LINKED ISSUE #3858: [Config] Document the 'database.hot_standby' option
**Product:** Tarantool
**Since:** 3.0
**Root document:** https://www.tarantool.io/en/doc/latest/reference/configuration/configuration_reference/#configuration-reference-database-hot-standby
**SME:** @ totktonada

# Details

Old content: https://www.tarantool.io/en/doc/latest/reference/configuration/#hot-standby
Dev issue: https://github.com/tarantool/tarantool/issues/775

### PR DESCRIPTION
**Pull request by andreyaksenov**
Fixes https://github.com/tarantool/doc/issues/3858.

Staging: https://docs.d.tarantool.io/en/doc/database-hot-standby/reference/configuration/configuration_reference/#configuration-reference-database-hot-standby

### CODE REVIEW

**Review by Totktonada — APPROVED** [review]
Thanks!

File: doc/reference/configuration/configuration_reference.rst [hunk 1]
```diff
@@ -1428,6 +1428,40 @@ The ``database`` section defines database-specific configuration parameters, suc
 
 .. confval:: database.hot_standby
 
+    Whether to start the server in the hot standby mode.
+    This mode can be used to provide failover without :ref:`replication <replication>`.
+
+    Suppose there are two cluster applications.
+    Each cluster has one instance with the same configuration:
```
Comment (p7nov) [inline doc/reference/configuration/configuration_reference.rst:1435]: I can't guess from this example the correct way to use this mode in multi-instance apps. Set it for all instances of two identical apps? Or maybe select a subset of instances to run in standby?
Not sure though if the deeper explanation is needed in the reference. Perhaps somewhere in a task-oriented page in the admin's guide (out of this PR's scope).

File: doc/reference/configuration/configuration_reference.rst [hunk 2]
```diff
@@ -1428,6 +1428,40 @@ The ``database`` section defines database-specific configuration parameters, suc
...
+    but fails because the primary instance has made a lock on this directory.
+
+    If the primary instance goes down for any reason, the lock is released.
+    In this case, the standby instance succeeds in taking the lock and becomes the primary instance.
+
+    ``database.hot_standby`` has no effect:
+
+    * If :ref:`wal.mode <configuration_reference_wal_mode>` is set to ``none``.
+    * If :ref:`wal.dir_rescan_delay <configuration_reference_wal_dir_rescan_delay>` is set to a large value on macOS or FreeBSD. On these platforms, the hot standby mode is designed so that the loop repeats every ``wal.dir_rescan_delay`` seconds.
+    * If spaces are created with :ref:`engine <space_opts_engine>` set to ``vinyl``.
```
Comment (p7nov) [inline doc/reference/configuration/configuration_reference.rst]: The old wording was clearer:
`for spaces created with engine = ‘vinyl’`

Now it's ambiguous:
- if _there are_ vinyl spaces?
- if _all_ spaces are vinyl?
- has no effect _only on_ vinyl spaces?

**Review by p7nov — APPROVED** [review]
LGTM 👍

File: doc/reference/configuration/configuration_reference.rst (hunk 1 above)
Comment (andreyaksenov) [reply doc/reference/configuration/configuration_reference.rst:1435]: > I can't guess from this example the correct way to use this mode in multi-instance apps.

The correct way is using `two cluster applications` with `each cluster having one instance`. And the GitHub examples at the end should help a reader. I agree that we can document this feature better. But, honestly, I don't think it is a very popular use case for Tarantool (maybe, I'm wrong).

> Set it for all instances of two identical apps? Or maybe select a subset of instances to run in standby?

This sounds like a feature request for a new configuration approach. For example, we can allow a user to add two instances, set `replication.failover` to smth like `hot_standby`, and provide the same path to WALs/snapshots for both instances. But still not sure that this scenario might be popular.  cc @Totktonada

**Pull request merged**